execution/engine.go

```diff
@@ -82,53 +85,61 @@ func (engine *PullEngine) Run(ctx context.Context) {
 
 	i := 0
 	for {
-		i++
-		if i%1 == 0 {
-			err := tx.Commit()
-			if err != nil {
-				log.Println("couldn't commit: ", err)
-				return
-			}
-			tx = engine.storage.BeginTransaction()
-		}
 		err := engine.loop(ctx, tx)
 		if err == ErrEndOfStream {
 			err := tx.Commit()
 			if err != nil {
-				log.Println("couldn't commit: ", err)
+				log.Println("engine: couldn't commit: ", err)
 			}
-			log.Println("end of stream, stopping loop")
+			log.Println("engine: end of stream, stopping loop")
 			return
 		}
 		if errors.Cause(err) == ErrNewTransactionRequired {
-			log.Println("engine new transaction required")
+			log.Println("engine: new transaction required")
 			err := tx.Commit()
 			if err != nil {
-				log.Println("couldn't commit: ", err)
-				return
+				log.Println("engine: couldn't commit: ", err)
 			}
+			engine.batchSizeManager.CommitSuccessful()
 			tx = engine.storage.BeginTransaction()
 		} else if waitableError := GetErrWaitForChanges(err); waitableError != nil {
+			log.Println("engine: listening for changes")
 			err := tx.Commit()
 			if err != nil {
-				log.Println("couldn't commit: ", err)
-				return
+				log.Println("engine: couldn't commit: ", err)
 			}
-			log.Println("engine listening for changes")
+			engine.batchSizeManager.CommitSuccessful()
 			err = waitableError.ListenForChanges(ctx)
 			if err != nil {
-				log.Println("couldn't listen for changes: ", err)
+				log.Println("engine: couldn't listen for changes: ", err)
 			}
+			log.Println("engine: received change")
 			err = waitableError.Close()
 			if err != nil {
-				log.Println("couldn't close listening for changes: ", err)
+				log.Println("engine: couldn't close listening for changes: ", err)
 			}
+			engine.batchSizeManager.Reset()
 			tx = engine.storage.BeginTransaction()
 		} else if err != nil {
 			tx.Abort()
-			log.Println(err)
+			log.Println("engine: ", err)
 			return // TODO: Error propagation? Add this to the underlying queue as an ErrorElement? How to do this well? Send it to the underlying IRS like a watermark?
 		}
+
+		i++
```

Comment:
What is this `i` really for? Could You change it's name cause it's not clear for me :<

Reply:
relic of the past